config.go

```diff
@@ -0,0 +1,231 @@
+package main
+
+import (
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ogier/pflag"
+	"github.com/papertrail/remote_syslog2/papertrail"
+	"github.com/papertrail/remote_syslog2/syslog"
+	"github.com/papertrail/remote_syslog2/utils"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	MIN_REFRESH_INTERVAL = RefreshInterval(10 * time.Second)
+	DEFAULT_CONFIG_FILE  = "/etc/log_files.yml"
+)
+
+type Config struct {
+	ConfigFile      string
+	Files           []string
+	DestHost        string
+	DestPort        int
+	Protocol        string
+	Hostname        string
+	RefreshInterval RefreshInterval
+	ExcludeFiles    RegexCollection
+	ExcludePatterns RegexCollection
+	LogLevels       string
+	DebugLogFile    string
+	PidFile         string
+	UseTCP          bool
+	UseTLS          bool
+	Daemonize       bool
+	Severity        syslog.Priority
+	Facility        syslog.Priority
+	Poll            bool
+	RootCAs         *x509.CertPool
+}
+
+type configfile struct {
+	Files       []string
+	Destination struct {
+		Host     string `yaml:"host"`
+		Port     int    `yaml:"port"`
+		Protocol string `yaml:"protocol"`
+	}
+	Hostname        string          `yaml:"hostname"`
+	RefreshInterval RefreshInterval `yaml:"new_file_check_interval"`
+	ExcludeFiles    RegexCollection `yaml:"exclude_files"`
+	ExcludePatterns RegexCollection `yaml:"exclude_patterns"`
+}
+
+func NewConfig() (*Config, error) {
+	self := &Config{
+		ConfigFile:      DEFAULT_CONFIG_FILE,
+		ExcludeFiles:    RegexCollection{},
+		ExcludePatterns: RegexCollection{},
+		DestHost:        "localhost",
+	}
+	if err := self.load(); err != nil {
+		return nil, err
+	}
+	// parse flags override config file
+	if err := self.override(); err != nil {
+		return nil, err
+	}
+	// check settings and set defaults if needed
+	if err := self.validate(); err != nil {
+		return nil, err
+	}
+	return self, nil
+}
+
+func (self *Config) load() error {
+	log.Infof("Reading configuration file %s", self.ConfigFile)
+	file, err := ioutil.ReadFile(self.ConfigFile)
+	// don't error if the default config file isn't found
+	if os.IsNotExist(err) && self.ConfigFile == DEFAULT_CONFIG_FILE {
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("Could not read the config file: %s", err)
+	}
+	config := &configfile{}
+	if err = yaml.Unmarshal(file, &config); err != nil {
+		return fmt.Errorf("Could not parse the config file: %s", err)
+	}
+	self.Files = config.Files
+	self.DestHost = config.Destination.Host
+	self.DestPort = config.Destination.Port
+	self.Protocol = config.Destination.Protocol
+	self.Hostname = config.Hostname
+	self.RefreshInterval = config.RefreshInterval
```

Comment:
I added a logging statement into main that prints out `config.RefreshInterval` and get:
```
2015-06-30 21:37:05 ERROR  remote_syslog.go:23 RefreshInterval: 30ns
```

Adding another logging statement into [SetYAML](https://github.com/papertrail/remote_syslog2/blob/issue-11/interval.go#L41) suggests it never gets called. I think we're using a slightly different version of the YAML lib, but in case it's useful, [this comment](https://github.com/papertrail/remote_syslog2/blob/master/config_manager.go#L32) is present in the current src.